runtime/src/iree/hal/drivers/vulkan/extensibility_util.h

```diff
@@ -67,6 +67,23 @@ iree_hal_vulkan_instance_extensions_t
 iree_hal_vulkan_populate_enabled_instance_extensions(
     const iree_hal_vulkan_string_list_t* enabled_extension);
 
+// Struct for miscellaneous device capabilities.
+typedef struct iree_hal_vulkan_device_capabilities_t {
+  // VkSubgroupFeatureFlags from VkPhysicalDeviceSubgroupProperties showing
```

Comment:
I feel this is going too detailed. I think we don't need all of these and can put some of them into the same bucket. We should think also from the llvmgpu perspective and how it maps to cuda and rocm.

Reply:
yeah, let's try to roll up the features into feature sets instead of granular features where we can so that we can share more code in the compiler (there will be times we can't, but if we can make a pass that produces code using subgroups that works with all SPIR-V targets (metal/d3d/vulkan/webgpu/etc) that'd be better than needing one pass per target (and one set of condition logic emission/etc). Basically, instead of a hal.device.vulkan.subgroup_operations, `hal.dispatch.subgroup_features` that could be common across targets.